tests/csapi/apidoc_room_event_test.go

```diff
@@ -0,0 +1,198 @@
+package csapi_tests
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+)
+
+func stringSliceContains(slice []string, s string) bool {
+	for _, value := range slice {
+		if value == s {
+			return true
+		}
+	}
+	return false
+}
+
+// sytest: Events come down the correct room
+func TestEventsInCorrectRoom(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	const (
+		roomAmount     = 10
+		goConcurrency  = 4
+		channelTimeout = 10 * time.Second
+	)
+
+	// Create all rooms
+
+	var doneRooms = make(chan string, roomAmount)
+	var todoRooms = make(chan bool, goConcurrency)
+
+	for i := 0; i < goConcurrency; i++ {
+		go func() {
+			for range todoRooms {
+				roomId := alice.CreateRoom(t, map[string]interface{}{
+					"preset": "public_chat",
+				})
+
+				doneRooms <- roomId
+			}
+		}()
+	}
+
+	for i := 0; i < roomAmount; i++ {
+		todoRooms <- true
+	}
+
+	close(todoRooms)
+
+	var rooms []string
+
+	for i := 0; i < roomAmount; i++ {
```

Comment:
I don't understand why we need to do this all concurrently. We're making 10 rooms, not 1000. Why not just:
```go
var rooms []string
for i := 0; i < roomAmount; i++ {
    rooms = append(rooms, alice.CreateRoom(t, map[string]interface{}{
        "preset": "public_chat",
    })
}
```
That is 6 lines and clearer that what is here.